actions/geolocation.js

```diff
@@ -0,0 +1,76 @@
+/*global navigator */
+'use strict';
+
+import { noop } from '../lib/utils'
+
+const hasNavigator = typeof navigator !== 'undefined';
+const geolocationIsSupported = hasNavigator ? ('geolocation' in navigator) : false;
+const PRE_ATTEMPT = 'GEOLOCATION_LOAD_PRE_ATTEMPT';
+const FAILURE = 'GEOLOCATION_LOAD_FAILURE';
+
+export function getCoordinates (context, payload, callback) {
+    callback = callback || noop;
+    navigator.geolocation.getCurrentPosition(position => {
+        callback(null, (position || {}).coords);
+    }, (err) => {
+        context.dispatch(FAILURE, err);
+        callback(err);
+    });
+}
+
+export function getCurrentUserLocation (context, payload, callback) {
+    callback = callback || noop;
+    context.dispatch(PRE_ATTEMPT);
+    getCoordinates(context, payload, function (err, coordinates) {
+        if (err) {
+            return callback(err);
+        }
+
+        coordinates = coordinates || {};
```

Comment:
small comment but maybe getCoordinates should return the default {} if not defined since its a helper anyways

Reply:
what do you mean?